lib/livebook/teams/connection.ex

```diff
@@ -106,7 +106,8 @@ defmodule Livebook.Teams.Connection do
         data = %{data | http_conn: conn, websocket: websocket}
         send(data.listener, {:connection_error, reason})
         :gen_statem.reply(from, {:error, reason})
-
+        Logger.warning("Teams WebSocket connection - send error: #{inspect(reason)}")
+        ensure_closed(data)
```

Comment:
I am not sure if that's necessary, but I imagine we can fail to send even if the connection is still open, so I added close attempts whenever we are about to open a new connection.